src/ProjectNeon/Assets/Features/Effects/EffectOnAttacked.cs

```diff
@@ -0,0 +1,29 @@
+﻿using UnityEngine;
+using System.Collections;
+
+/**
+ * Effects that trigger to the target of an attack upon an attack event
+ */
+public class EffectOnAttacked : Effect
+{
+    private Effect effect;
+    private Member performer;
+    private Target effectTarget;
+
+    public EffectOnAttacked(Effect shield)
+    {
+        this.effect = shield;
+        BattleEvent.Subscribe<Attack>(attack => Execute(attack), this);
+    }
+
+    void Effect.Apply(Member source, Target target)
+    {
+        this.performer = source;
+        this.effectTarget = target;
+    }
+
+    void Execute(Attack attack)
+    {
+        this.effect.Apply(this.performer, this.effectTarget);
```

Comment:
This effect is not correctly scoped. This will apply if the Character who has this effect is not the one being targeted by an attack.

Reply:
@SilasReinagel Right, I forgot this situation